src/openzaak/components/catalogi/api/viewsets/zaaktype.py

```diff
@@ -84,3 +91,25 @@ class ZaakTypeViewSet(
         "publish": SCOPE_ZAAKTYPES_WRITE,
     }
     concept_related_fields = ["besluittype_set"]
+
+    @swagger_auto_schema(request_body=no_body)
+    @action(detail=True, methods=["post"])
+    def publish(self, request, *args, **kwargs):
+        instance = self.get_object()
+
+        # check related objects
+        besluittypen = instance.besluittype_set.all()
+        informatieobjecttypen = instance.heeft_relevant_informatieobjecttype.all()
+
+        for types in [besluittypen, informatieobjecttypen]:
+            for relative_type in types:
+                if relative_type.concept:
+                    msg = _("All relative resources should be published")
+                    raise PermissionDenied(detail=msg)
```

Comment:
Should be 400 response

Reply:
PermissionDenied is replaced with ValidationError